homeassistant/components/fibaro/climate.py

```diff
@@ -0,0 +1,276 @@
+"""Support for Fibaro thermostats."""
+import logging
+
+from homeassistant.components.climate.const import (
+    STATE_AUTO, STATE_COOL,
+    STATE_DRY, STATE_FAN_ONLY, STATE_HEAT,
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_OPERATION_MODE, SUPPORT_FAN_MODE)
+
+from homeassistant.components.climate import (
+    ClimateDevice)
+
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    STATE_OFF,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT)
+
+from . import (
+    FIBARO_DEVICES, FibaroDevice)
+
+SPEED_LOW = 'low'
+SPEED_MEDIUM = 'medium'
+SPEED_HIGH = 'high'
+STATE_AUXILIARY = 'auxiliary'
+STATE_RESUME = 'resume'
+STATE_MOIST = 'moist'
+STATE_AUTO_CHANGEOVER = 'auto changeover'
+STATE_ENERGY_HEAT = 'energy heat'
+STATE_ENERGY_COOL = 'energy cool'
+STATE_FULL_POWER = 'full power'
+STATE_FORCE_OPEN = 'force open'
+STATE_AWAY = 'away'
+FAN_AUTO_HIGH = 'auto_high'
+FAN_AUTO_MEDIUM = 'auto_medium'
+FAN_CIRCULATION = 'circulation'
+FAN_HUMIDITY_CIRCULATION = 'humidity_circulation'
+FAN_LEFT_RIGHT = 'left_right'
+FAN_UP_DOWN = 'up_down'
+FAN_QUIET = 'quiet'
+STATE_FURNACE = 'furnace'
+
+DEPENDENCIES = ['fibaro']
+
+_LOGGER = logging.getLogger(__name__)
+
+# SDS13781-10 Z-Wave Application Command Class Specification 2019-01-04
+# Table 128, Thermostat Fan Mode Set version 4::Fan Mode encoding
+FANMODES = {
+    0: STATE_OFF,
+    1: SPEED_LOW,
+    2: FAN_AUTO_HIGH,
+    3: SPEED_HIGH,
+    4: FAN_AUTO_MEDIUM,
+    5: SPEED_MEDIUM,
+    6: FAN_CIRCULATION,
+    7: FAN_HUMIDITY_CIRCULATION,
+    8: FAN_LEFT_RIGHT,
+    9: FAN_UP_DOWN,
+    10: FAN_QUIET,
+    128: STATE_AUTO
+}
+
+# SDS13781-10 Z-Wave Application Command Class Specification 2019-01-04
+# Table 130, Thermostat Mode Set version 3::Mode encoding.
+OPMODES = {
+    0: STATE_OFF,
+    1: STATE_HEAT,
+    2: STATE_COOL,
+    3: STATE_AUTO,
+    4: STATE_AUXILIARY,
```

Comment:
Only operation states present in the base climate component are allowed as operation modes.

Please limit the home assistant modes to those states.

Reply:
Done